detector/muonSystem/muonSystemMuRWELL_o1_v01.cpp

```diff
@@ -0,0 +1,946 @@
+
+  dd4hep::Position barrelTrans(0., 0., 0.);
+  dd4hep::PlacedVolume barrelPhys = detectorVolume.placeVolume(BarrelVolume, dd4hep::Transform3D(dd4hep::RotationZ(0.), barrelTrans));
+  barrelPhys.addPhysVolID("type", 0); 
+  dd4hep::DetElement BarrelDE(detElement, "MSBarrelDE" , 0);
+  BarrelDE.setPlacement(barrelPhys);
+  BarrelVolume.setVisAttributes(lcdd.visAttributes("no_vis"));
+
+  // ---------- loop to creat the barrel layers -----------
+
+  for (int numBarrelLayer = 0; numBarrelLayer < numBarrelDetectorLayers; ++numBarrelLayer){
+
+  double barrelLayerRMin = radius + numBarrelLayer * (2 * detectorVolumeThickness) + numBarrelLayer * barrelRadiatorThickness;   // Automation of inner radius of different layers, taking into account that every detector layer is followed by a yoke(radiator) layer.
+  double barrelLayerRMax = barrelLayerRMin + 2 * detectorVolumeThickness;
+  double barrelLayerRMid = (barrelLayerRMin + barrelLayerRMax)/ 2.0;
+  double barrelLayerLength;
+  if (numBarrelLayer == numBarrelDetectorLayers - 1){
+   barrelLayerLength = barrelLength + (numEndcapDetectorLayers * 2) * (2 * detectorVolumeThickness) + (numEndcapRadiators * 2) * endcapRadiatorThickness; // This condition to make the last barrel layer encloses all the endcap layers inside it.
+  } else {
+   barrelLayerLength = barrelLength;
+  }
+
+  dd4hep::PolyhedraRegular  BarrelDetectorLayer(numSides, barrelLayerRMin, barrelLayerRMax, barrelLayerLength);
+  std::string barrelDetectorName = dd4hep::xml::_toString(numBarrelLayer +1, "MS-BarrelDetectorLayer%d");
+  dd4hep::Volume BarrelDetectorLayerVolume(barrelDetectorName, BarrelDetectorLayer, mat);
+
+  // sideLength and sideLength2 are the lengths of the parallel sides of the trapezoid.
+  double sideLength = 2 * (barrelLayerRMid - detectorVolumeThickness/2.0) * std::tan(shapeAngle_radians);
+  double sideLength2 = 2 * (barrelLayerRMid + detectorVolumeThickness/2.0) * std::tan(shapeAngle_radians);  
+  double sideEnvX = detectorVolumeThickness/2.0;
+  double sideEnvY = (sideLength / 2.0); 
+  double sideEnvY2 = (sideLength2 / 2.0);   
+  double sideEnvZ = (barrelLayerLength / 2.0); 
+  double remainderZ = std::fmod((barrelLayerLength - 2 * clearance), (2 * dimensions.z() - overlapZ)) / (2 * dimensions.z()) - (2 * clearance / dimensions.z());  
+ 
+  for (int side = 0; side < numSides; ++side) {
+      
+      int sideID = (numBarrelLayer + 1) * 100 + (side +1);  // to differentiated with the same side in different layers.
+      dd4hep::Box sideEnvelope(sideEnvX, sideEnvY, sideEnvZ);
+      dd4hep::Box sideEnvelope2(sideEnvX, sideEnvY2, sideEnvZ);      
+      std::string sideName = dd4hep::xml::_toString(sideID, "side%d");
+      dd4hep::Volume sideVol(sideName, sideEnvelope, mat);
+      dd4hep::Volume sideVol2(sideName, sideEnvelope2, mat);      
+
+      double angle_degrees = shapeAngle * side; // Calculate the angle for each side
+      double angle_radians = angle_degrees * M_PI / 180.0;
```

Comment:
to avoid confusion I would suggest to work on angle natural units of dd4hep. In addition, conversion of deg <-> rad can be done using ROOT factos`TMath::DegToRad()`